### DISCUSSION TITLE
Embed an Image into Excel ("Place in Cell")

### DISCUSSION THREAD

**Discussion by Ephraim-Bryski** [exceljs/exceljs#2839] [Q&A]
I'm looking to embed an image in a cell of an Excel file. The image should be locked into the cell, preventing the user from freely moving it around, and it should automatically resize when the user adjust the cell size; this is Excel's "Place in Cell" option, instead of "Place over Cell". In issue #1029 , Knight-H provided the solution:

```javascript
sheet.addImage(imageId, `${cellId}:${cellId}`)
```

This makes an image fit the size of the cell, which almost works, but it is still placed over the cells, not inside.

Another approach :
```javascript
sheet.addImage(imageId, {
	tl: { col: 1, row: 1 },
	ext: { width: 100, height: 100 },
})
```

has the same issue. (Setting `editAs` to "oneCell", has no effect as far as I can see).

Is this functionality currently possible in ExcelJS? If not, are there other node.js libraries that support it, or is there interest in adding this option?

**Comment by zjtakato** [comment]
any news？

**Comment by barbaldo** [comment]
This would be SO useful
